### PR TITLE
Path: Keep windows flag during canonicalize (fixes #102)

### LINKED ISSUE #102: wrong org.eclipse.core.runtime.Path on windows if "..\.." is appended
The contract of Path is that windows`"\"` should be replaced by `"/"`. However the windows flag is lost when ascending to root path with `".."` therefore further appends on such a path are not using canonical `"/"` anymore. They use` "\"` path separator instead:
```

assertEquals("4.2XY.win", Path.forWindows("/").append("x\\y"),
				Path.forWindows("/").append("x").append("y").append("..\\..").append("x\\y"));

junit.framework.AssertionFailedError: 4.2XY.win expected:</x/y> but was:</x\y>
```

### PR DESCRIPTION
**Pull request by jukzi**
The WINDOWS flag was lost when escaping to root path with appending ".."

### CODE REVIEW

File: bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/Path.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2000, 2015 IBM Corporation and others.
+ * Copyright (c) 2000, 20122 IBM Corporation and others.
```
Comment (bjhargrave) [inline bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/Path.java:2]: Year is in the future 😀

Comment (jukzi) [reply bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/Path.java]: :-)

**Comment by jukzi** [comment]
unrelated build errors:

Some blocking (most likely link/compilation) errors are present:
[FATAL] File CachingService.java at line 137: Bundle cannot be resolved to a type (location: /home/runner/work/equinox/equinox/bundles/org.eclipse.equinox.weaving.caching.j9/src/org/eclipse/equinox/weaving/internal/caching/j9/CachingService.java)
[FATAL] File CachingService.java at line 29: The import org.eclipse.equinox.service cannot be resolved (location: /home/runner/work/equinox/equinox/bundles/org.eclipse.equinox.weaving.caching.j9/src/org/eclipse/equinox/weaving/internal/caching/j9/CachingService.java)

**Review by bjhargrave — APPROVED** [review]

**Comment by mickaelistria** [comment]
@jukzi Please bump bundle version by +0.0.100 by the way

**Comment by mickaelistria** [comment]
Thanks for the patch and @bjhargrave for the review!

**Pull request merged**